### ISSUE TITLE
Component: p-Dropdown placeholder/values not displayed

### ISSUE THREAD

**Issue by asithade** [primefaces/primeng#14695]
### Describe the bug

When using p-dropdown, we've noticed that some of the dropdowns when a value is not selected end up with a short height with no placeholder value set. The weird thing is that not all p-dropdowns are affected -- it's quite random on how it is chosen. 

* Some have dynamic placeholders (#14320) and some have static placeholders. 
* Could be related to dropdowns with dynamic values too
* Some have values set, but the input doesn't display it
  * Only after selecting a different value does it reflect on the input

https://github.com/primefaces/primeng/assets/3814354/de25971c-1a7d-4c3a-a603-2742c39ce1d2

https://github.com/primefaces/primeng/assets/3814354/52268d82-2d2c-47cc-b845-607aec15f341

### Environment

I'm expecting the dropdown to either display the current value or the placeholder. 

### Reproducer

[https://stackblitz.com/edit/github-lobmlu?file=src%2Fapp%2Fapp.component.html](https://stackblitz.com/edit/github-lobmlu?file=src%2Fapp%2Fapp.component.html)

### Angular version

17.1.1

### PrimeNG version

17.4.0 & 17.5.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.19.0

### Browser(s)

Brave Version 1.62.153 Chromium: 121.0.6167.85 

### Expected behavior

I expected the dropdown to either show the placeholder value or the active value

**Comment by stepkillah** [comment]
Same issue, noticed after updating to 17.5

**Comment by mdybich** [comment]
Same issue here, works as expected on 17.4.0 but it's broken on 17.5.0

**Comment by Draganlazarev90** [comment]
Also doesn't work with reactive forms, when you set fromcontrol value it doesn't display it

**Comment by mastersys-giovanni** [comment]
Same error.

**Comment by buchatsky** [comment]
Surprise: it does not work in my project, but it does work in [Stackblitz](https://stackblitz.com/edit/qmqtnu).
Race condition? Nested components?

**Comment by jozif001** [comment]
same issue after upgrading to primeng version 17.5.0,
using Angular 17.1.0

**Comment by Cr3aHal0** [comment]
Hello everyone,
[asithade](https://github.com/asithade) I tried binding a formControl or ngModel to your p-dropdown and it seems to make the placeholder appear, same for [buchatsky](https://github.com/buchatsky) example. This is indeed not a fix but it can temporary help you having a proper rendering while PrimeNG team has time to fix it.

**Comment by davidediak** [comment]
yikes this is so bad

**Comment by cetincakiroglu** [comment]
Hi,

Thanks a lot for letting me know, the issue is fixed now.

**Comment by mdybich** [comment]
@cetincakiroglu The issue should be closed when the version is released, not when the commit is merged to master branch.

**Comment by asithade** [comment]
Works now on the latest release. Thank you!

**Comment by Sothecom** [comment]
Hi, this is no longer working in 17.7, should I raise a new issue?

**Comment by pradnya-khairnar** [comment]
same issue facing in Primeng 17.18.9. using formControl/ngModel doesn't set value in dropdown textbox after changing value.

**Comment by Medioman92** [comment]
Still broken in 17.18.11. though adding a placeholder "fixes" the problem

**Comment by Yoranium238** [comment]
The issue appears to be a problem where the bound input (ngModel) is an empty string ("").  The placeholder will only display if the bound input is undefined.

html

<div class="card flex justify-content-center">
  <p-dropdown [(ngModel)]="dropdownString" [editable]="true" placeholder="Dropdown String"></p-dropdown>
</div>

component
export class AutocompleteDropdownDemoComponent implements OnInit {
  countries: any[] | undefined;

  selectedCountry: any;
  dropdownString: string;             // Sets placeholder
  dropdownString: string = "";     // Does not set placeholder
